site/docs/getting-started/tutorials/postgresql_cdc_to_snowflake.md

```diff
@@ -0,0 +1,449 @@
+);
+
+CREATE TABLE IF NOT EXISTS public.flow_watermarks (slot TEXT PRIMARY KEY, watermark TEXT);
+GRANT ALL PRIVILEGES ON TABLE public.flow_watermarks TO flow_capture;
+CREATE PUBLICATION flow_publication;
+ALTER PUBLICATION flow_publication SET (publish_via_partition_root = true);
+ALTER PUBLICATION flow_publication ADD TABLE public.flow_watermarks, public.products;
+```
+
+In the `init.sql` file, you create the products table and all the database objects Flow requires for real-time CDC streaming.
+
+
+### Configuring PostgreSQL for CDC<a id="configuring-postgresql-for-cdc"></a>
+
+To enable CDC replication in PostgreSQL, several database objects need to be created and configured. These objects facilitate the capture and propagation of data changes to downstream systems. Let's examine each object and its significance in the context of CDC replication:
+
+```sql
+CREATE USER flow_capture WITH PASSWORD 'secret' REPLICATION;
+```
+
+This user is dedicated to the CDC replication process. It is granted the necessary permissions to read all data from the database, allowing it to capture changes across tables efficiently. In a production environment, make sure you use a more secure password than what is in the example.
+
+```sql
+GRANT pg_read_all_data TO flow_capture;
+```
+
+Granting the `pg_read_all_data` privilege to the `flow_capture` user ensures that it can access and read data from all tables in the database, essential for capturing changes.
+
+```sql
+CREATE TABLE products (...)
+```
+
+The source tables, such as the `products` table in this example, contain the data whose changes we want to capture and replicate. These tables must have a primary key defined to uniquely identify rows.
+
+```sql
+CREATE TABLE IF NOT EXISTS public.flow_watermarks (...)
+```
+
+The `flow_watermarks` table is used to store metadata about the replication process, specifically the position of the replication slot and the corresponding watermark. This information helps track the progress of replication and ensures consistency.
+
+```sql
+GRANT ALL PRIVILEGES ON TABLE public.flow_watermarks TO flow_capture;
+```
+
+The `flow_capture` user needs full privileges on the `flow_watermarks` table to insert, update, and query metadata related to the replication process.
+
+```sql
+CREATE PUBLICATION flow_publication;
+ALTER PUBLICATION flow_publication SET (publish_via_partition_root = true);
+ALTER PUBLICATION flow_publication ADD TABLE public.flow_watermarks, public.products;
+```
+
+A publication defines a set of tables whose changes will be replicated. In this case, the `flow_publication` publication includes the `public.flow_watermarks` and `public.products` tables.
+
+These commands configure the `flow_publication` publication to publish changes via partition root and add the specified tables to the publication. By setting `publish_via_partition_root` to true, the publication ensures that updates to partitioned tables are correctly captured and replicated.
```

Comment:
Might add a note "The table in this tutorial is not partitioned, but we recommend always setting `publish_via_partition_root` when creating a publication"

Reply:
👍